src/OpenTelemetry.ClrProfiler.Managed/Configuration/EnvironmentConfigurationHelper.cs

```diff
@@ -70,21 +71,32 @@ private static TracerProviderBuilder SetExporter(this TracerProviderBuilder buil
                     break;
                 case "otlp":
 #if NETCOREAPP3_1
-                    // Adding the OtlpExporter creates a GrpcChannel.
-                    // This switch must be set before creating a GrpcChannel/HttpClient when calling an insecure gRPC service.
-                    // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
-                    AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+                    if (!settings.OtlpExportProtocol.HasValue || settings.OtlpExportProtocol != OtlpExportProtocol.HttpProtobuf)
+                    {
+                        // Adding the OtlpExporter creates a GrpcChannel.
+                        // This switch must be set before creating a GrpcChannel/HttpClient when calling an insecure gRPC service.
+                        // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
+                        AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
```

Comment:
That's an interesting idea. I suppose that you are more interested in detecting the usage of `http` instead of `https`, as opposed to some other scheme like `grpc` or a named pipe? If that's the case, perhaps we should add that as a note to #217? 

When taken within the context of #217, do we even need all of those other checks in the `if` statement if we are requiring someone to set an environment variable to opt-in to enabling that global switch?

Reply:
> When taken within the context of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/217, do we even need all of those other checks in the if statement if we are requiring someone to set an environment variable to opt-in to enabling that global switch?

I agree here with @nrcventura

Reply:
Yeah, I forgot about #217. Having the setting as an opt-in is a better option.